api/pkg/s3/service.go

```diff
@@ -160,3 +161,36 @@ func HandleS3Error(response *restful.Response, request *restful.Request, err err
 
 	return nil
 }
+
+func (s *APIService) getBackendFromTier(ctx context.Context, tierName string) string {
+	log.Info("The received tier name for getting backend name:", tierName)
+	var backendId, backendName string
+
+	tierResp, err := s.backendClient.ListTiers(ctx, &backend.ListTierRequest{})
+	if err != nil {
+		log.Error("list tier failed during getting backends from tier")
+		return ""
+	} else {
+		if len(tierResp.Tiers) > 0 {
+			for _, tier := range tierResp.Tiers {
+				if tier.Name == tierName {
+					backendId = tier.Backends[rand.Intn(len(tier.Backends))]
+					break
+				}
+			}
+		}
+	}
+
+	if backendId != "" {
+		backendRep, err := s.backendClient.GetBackend(ctx, &backend.GetBackendRequest{Id: backendId})
+		if err != nil {
+			log.Error("the selected backends from tier doesn't exists.")
```

Comment:
Will this error be reached if we have already validated backend while creating Tier? There cab be error due to some other issues

Reply:
There could be chance, admin removed this backendsjust before reaching this line